packages/messaging/lib/index.d.ts

```diff
@@ -839,51 +815,12 @@ export namespace FirebaseMessagingTypes {
      * unsubscribe();
      * ```
      *
+     * NOTE: this appears to be Android-only as of firebase-ios-sdk v7+
+     *
      * @param listener Called when the FCM deletes pending messages.
      */
     onDeletedMessages(listener: () => void): () => void;
```

Comment:
This is part of the upstream messaging API which we're removing, given that the user can no longer `sendMessage` this will never trigger, so can be removed.

Reply:
Ah - just missed it, I wasn't sure about onDeletedMessages until further along in the forward port and this was an orphan remnant yes, thanks

Reply:
per discussion on slack, these still exist and appear unrelated to upstream send, however they are on iOS as well so my documentation note here is not correct. I will resolve

Reply:
On even more discussion, direct channel (and thus upstream send + onDeletedMessages) are completely gone on iOS but they still exist on Android. This commit will be fundamentally reshaped to leave it intact for Android-only, with a more complete purge on iOS.